Neos.ContentRepository.Core/Classes/SharedModel/Privilege/Privileges.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Privilege;
+
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+
+/**
+ * @api
+ */
+final class Privileges
+{
+    private function __construct(
+        public readonly ?ContentStreamPrivilege $contentStreamPrivilege,
+
+    ) {
+    }
+
+    public static function create(): self
+    {
+        return new self(null);
+    }
+
+    public function with(
+        ContentStreamPrivilege $contentStreamPrivilege = null,
+    ): self
+    {
+        return new self(
+            $contentStreamPrivilege ?? $this->contentStreamPrivilege,
+        );
+    }
+
+    public function isContentStreamAllowed(ContentStreamId $contentStreamId): bool
+    {
+        if ($this->contentStreamPrivilege === null) {
+            return true;
+        }
+        return $this->contentStreamPrivilege->allowedContentStreamIds->contain($contentStreamId);
```

Comment:
Hm, I am not so sure about this one - IMHO this must be done lazily (otherwise we could under some scenarios have REALLY long lists of contentStreamIds; where only a single one will be relevant here)

Reply:
isn't it just the currently active Content Streams of the user workspace (and its base workspace(s))?